File: abc324b/abc324b.hs
import Data.Bool (bool)

main :: IO ()
main = solve <$> readLn >>= putStrLn

solve :: Int -> String
solve = bool "No" "Yes" . (== 1) . divn 2 . divn 3

divn n x
  | x `mod` n == 0 = divn n (x `div` n)
  | otherwise= x

File: abc324b/abc324b.ml
open Printf
open Scanf

let solve n =
  let rec divn n x =
    if x mod n = 0 then divn n (x / n) else x in
  if divn 3 (divn 2 n) = 1 then "Yes" else "No"

let () =
  scanf "%d " solve |> printf "%s\n"
